display_list/display_list_mask_filter.h

```diff
@@ -58,12 +58,13 @@ class DlMaskFilter
 // filter is then used to combine those colors.
 class DlBlurMaskFilter final : public DlMaskFilter {
  public:
-  DlBlurMaskFilter(SkBlurStyle style, SkScalar sigma)
-      : style_(style), sigma_(sigma) {}
+  DlBlurMaskFilter(SkBlurStyle style, SkScalar sigma, bool respect_ctm = true)
+      : style_(style), sigma_(sigma), respect_ctm_(respect_ctm) {}
   DlBlurMaskFilter(const DlBlurMaskFilter& filter)
-      : DlBlurMaskFilter(filter.style_, filter.sigma_) {}
+      : DlBlurMaskFilter(filter.style_, filter.sigma_, filter.respect_ctm_) {}
   DlBlurMaskFilter(const DlBlurMaskFilter* filter)
-      : DlBlurMaskFilter(filter->style_, filter->sigma_) {}
+      : DlBlurMaskFilter(filter->style_, filter->sigma_, filter->respect_ctm_) {
+  }
```

Comment:
You will need to modify the code in `DisplayListBoundsCalculator::AdjustBoundsForPaint` that computes bounds of rendering operations that adjust the bounds for a MaskFilter. It gets complicated because the bounds are being padded before adjusting for the CTM which is done outside (after) that function, so I'm not sure how to pad them appropriately for device pixels (another reason why I really think we need to ditch this flag).

Reply:
The old implementation was delegating to Skia's `SkBlurMaskFilterImpl::computeFastBounds`, which was ignoring the respectCTM flag.

So the behavior of this implementation should be equivalent to what the engine is currently doing.

Reply:
Filed https://github.com/flutter/flutter/issues/115114 about fixing the bounds and https://github.com/flutter/flutter/issues/115112 about potentially deleting this feature in the future.